### DISCUSSION TITLE
Support for even more divisions?

### DISCUSSION THREAD

**Discussion by dbarati-bf** [rxhanson/Rectangle#564] [Ideas]
Will there be support for even more divisions at some point? I really love the program, but I see that for a high resolution 49 inch ultra wide screen, even 8th divisions would be quite useful (similar to today's 6th but even more narrow). 

I guess if there was a command for halving the window size vertically, it would be possible to do 8ths by executing the existing command for 4th and then a command for halving the window.

**Comment by rxhanson** [comment]
The halving behavior that you’re asking for can be enabled using the terminal command in the following readme section and restarting the app 
https://github.com/rxhanson/Rectangle#resize-on-directional-move

I have plans to make it easier for developers to add more window sizes in a future release. 

Beyond that, you could use the custom shortcut builder in [Rectangle Pro](https://rectangleapp.com/pro) (paid).

**Reply by dbarati-bf** [reply]
That’s perfect, thanks! I didn’t know about the terminal configurations! And it works, but only sometimes.

Here’s what I set up:

```
defaults write com.knollsoft.Rectangle halveHeightUp    -dict-add keyCode -float 35 modifierFlags -float 1835305
defaults write com.knollsoft.Rectangle halveHeightDown  -dict-add keyCode -float 41 modifierFlags -float 1835305
defaults write com.knollsoft.Rectangle halveWidthLeft   -dict-add keyCode -float 37 modifierFlags -float 1835305
defaults write com.knollsoft.Rectangle halveWidthRight  -dict-add keyCode -float 39 modifierFlags -float 1835305
```

If I have a fullscreen window, all of these commands work as expected.

However, coming back to my original request: if I put a window in First Quarter (25% of the width, full height) and then follow it up with halveHeightUp, it doesn’t work. This isn’t the only combination where the halving commands don’t trigger.

Any idea why? It feels like there may be constraints on the minimum size or shape of a window that prevent these four commands from applying.

**Reply by dbarati-bf** [reply]
Actually, I just noticed that you have eights supported now (also in the terminal options). E.g. topLeftEighth with repeat on command solves my original problem.
